tests/libvmifact/factory.go

```diff
@@ -49,16 +50,19 @@ func NewFedora(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
 
 // NewCirros instantiates a new CirrOS based VMI configuration
 func NewCirros(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
-	// Supplied with no user data, Cirros image takes 230s to allow login
-	withNonEmptyUserData := libvmi.WithCloudInitNoCloudEncodedUserData("#!/bin/bash\necho hello\n")
-
 	cirrosOpts := []libvmi.Option{
 		libvmi.WithContainerDisk("disk0", cd.ContainerDiskFor(cd.ContainerDiskCirros)),
-		withNonEmptyUserData,
 		libvmi.WithResourceMemory(cirrosMemory()),
 	}
 	cirrosOpts = append(cirrosOpts, opts...)
-	return libvmi.New(cirrosOpts...)
+	vmi := libvmi.New(cirrosOpts...)
+
+	// Supplied with no user data, Cirros image takes 230s to allow login
+	if !cloudInitNoCloudExists(vmi.Spec.Volumes) {
```

Comment:
I think this is not enough. We don't want to add the user data also in case where one of the user data fields in the `CloudInitConfigDrive` is not empty.

@EdDev

Reply:
I think this fits to the existing behavior, i.e. one can define both in parallel and one will not override the other.

If using the `ConfigDrive` helps with the "quick" boot, then we can add it as a condition as well. If it does not help with that, then it should not be added. But this logic did not existed before per what I know.

Reply:
The previous logic is less important, because the purpose of this work is to move more old code to use libvmi. We'll need to support more cases now, specifically new calls to NewCirros, that will now fail.